Scripts/Runtime/Entities/ProxyDataStream/Data/ProxyDataWrapper.cs

```diff
@@ -2,41 +2,42 @@
 using Unity.Collections;
 using Unity.Entities;
 
-namespace Anvil.Unity.DOTS.Data
+namespace Anvil.Unity.DOTS.Entities
 {
-    internal readonly struct PDWrapper<T> : IEquatable<PDWrapper<T>>
-        where T : unmanaged, IEntityProxyData
+    internal readonly struct ProxyDataWrapper<TData> : IEquatable<ProxyDataWrapper<TData>>
+        where TData : unmanaged, IProxyData
     {
-        public static bool operator==(PDWrapper<T> lhs, PDWrapper<T> rhs)
+        public static bool operator ==(ProxyDataWrapper<TData> lhs, ProxyDataWrapper<TData> rhs)
         {
             //Note that we are not checking if the Payload is equal because the wrapper is only for origin and lookup
             //checks. 
```

Comment:
It's the second one. 

There SHOULD never be a circumstance where two pieces of data share the same ID but have different Payloads. That would lead to ambiguity and the whole program would fall apart. 

I've added an expensive check to this due to `TData`'s double boxing in using the `.Equals` function. 

We'll have the safety now if `ANVIL_DEBUG_SAFETY_EXPENSIVE` is turned on.